qoffer-domain/src/test/groovy/life/qbic/portal/portlet/customers/search/SearchCustomerTest.groovy

```diff
@@ -0,0 +1,55 @@
+package life.qbic.portal.portlet.customers.search
+
+import life.qbic.datamodel.dtos.business.AcademicTitle
+import life.qbic.datamodel.dtos.business.Affiliation
+import life.qbic.datamodel.dtos.business.Customer
+import life.qbic.portal.portlet.SearchCriteria
+import life.qbic.portal.portlet.exceptions.DatabaseQueryException
+import spock.lang.Specification
+
+
+/**
+ * Test the search of customers
+ *
+ * Stubs the database connection and verifies that if a customer is found the {@link SearchCustomerOutput} is verified or if not
+ * an exception leads to a failure notification
+ *
+ * @since: 1.0.0
+ *
+ */
+class SearchCustomerTest extends Specification{
+
+
+    def "find a searched customer"(){
+        given:
+        SearchCustomerOutput output = Mock(SearchCustomerOutput.class)
+        SearchCustomerDataSource ds = Stub(SearchCustomerDataSource.class)
+        SearchCustomer searchCustomer = new SearchCustomer(output,ds)
+
+        Customer luke = new Customer("Luke","Skywalker", AcademicTitle.DOCTOR, "a.b@c.de", new ArrayList<Affiliation>())
+
+        ds.findCustomer(_ as SearchCriteria) >> [luke]
+
+        when:
+        searchCustomer.searchCustomer("Luke","Skywalker")
+
+        then:
+        1* output.successNotification("Found 1 customers matching Luke Skywalker")
+    }
+
+    def "throw an exception if a customer is not found"(){
+        given:
+        SearchCustomerOutput output = Mock(SearchCustomerOutput.class)
+        SearchCustomerDataSource ds = Stub(SearchCustomerDataSource.class)
+        SearchCustomer searchCustomer = new SearchCustomer(output,ds)
+
+        ds.findCustomer(_ as SearchCriteria) >> {throw new DatabaseQueryException("Customer not found")}
```

Comment:
hmm... I would keep it since we then now it's the error for the database. And if the database would change we would still need to update our domain since it throws the error, right?

Reply:
I don't think I get what you are saying here.